### DISCUSSION TITLE
How to pass current buffer as argument to a shell script?

### DISCUSSION THREAD

**Discussion by lando-tech** [helix-editor/helix#11899] [Q&A] [answered]
I am trying to write a shell script that takes the current file/buffer as an argument. 
A simple example: 

`#!/bin/bash python3  "$1" # Argument is the path to the current buffer`
 
Is there a way to achieve this using a custom key binding in the config.toml?

Here is the current configuration I was experimenting with: 

`[keys.normal.'\']
R = ":run-shell-command ~/<path-to-script> {file-path}"`

I just started using helix and I am still reading through the docs, but I haven't found anything yet on this particular feature or if it even exists.

**Comment by quantonganh** [comment]
Do you want to give [WezTerm](https://wezfurlong.org/wezterm/index.html) a try? https://quantonganh.com/2023/08/19/turn-helix-into-ide.md

**Comment by david-crespo** [comment, marked as answer]
The [command expansions PR](https://github.com/helix-editor/helix/pull/11164) adds this functionality. It's not merged yet, so you would need to build from source to use it. You can see an example in the [docs](https://github.com/helix-editor/helix/blob/c0666d1219a82cd4e850a30ca1a7f18294c58e9a/book/src/commands.md) part of the PR:

```toml
[keys.normal]
# Print blame info for the line where the main cursor is.
C-b = ":echo %sh{git blame -L %{linenumber} %{filename}}"
```

The PR still needs maintainer approval, but I've been building from source and using that PR for months. It works well.

**Reply by lando-tech** [reply]
Thanks! That did the trick.

**Reply by wmstack** [reply]
What happens when the buffer isn't backed by a file? Would one still by able to supply it as a file handle?
